digits/extensions/view/imageSegmentation/view.py

```diff
@@ -134,24 +171,68 @@ def process_data(self, input_id, input_data, output_data):
         """
         # assume the only output is a CHW image where C is the number
         # of classes, H and W are the height and width of the image
-        data = output_data[output_data.keys()[0]].astype('float32')
+        class_data = output_data[output_data.keys()[0]].astype('float32')
         # retain only the top class for each pixel
-        data = np.argmax(data,axis=0)
+        class_data = np.argmax(class_data, axis=0).astype('uint8')
 
         # remember the classes we found
-        found_classes = np.unique(data)
+        found_classes = np.unique(class_data)
```

Comment:
do we still need `self.found_classes` below? Now that we have one legend per image I think this can go away.